lib/smack/SmackRep.cpp

```diff
@@ -98,6 +171,16 @@ string SmackRep::memReg(unsigned idx) {
   return s.str();
 }
 
+string SmackRep::memReg(unsigned idx, bool safety, unsigned type) {
```

Comment:
Good idea. But let's call this method `memPath` for the moment instead of `memName` to be slightly more descriptive.